ecosystem/gstreamer_plugin/gst_mtl_st30p_rx.c

```diff
@@ -0,0 +1,643 @@
+    case PROP_RX_DEV_ARGS_SIP:
+      g_value_set_string(value, src->devArgs.local_ip_string);
+      break;
+    case PROP_RX_DEV_ARGS_DMA_DEV:
+      g_value_set_string(value, src->devArgs.dma_dev);
+      break;
+    case PROP_RX_PORT_PORT:
+      g_value_set_string(value, src->portArgs.port);
+      break;
+    case PROP_RX_PORT_IP:
+      g_value_set_string(value, src->portArgs.session_ip_string);
+      break;
+    case PROP_RX_PORT_UDP_PORT:
+      g_value_set_uint(value, src->portArgs.udp_port);
+      break;
+    case PROP_RX_PORT_PAYLOAD_TYPE:
+      g_value_set_uint(value, src->portArgs.payload_type);
+      break;
+    case PROP_RX_PORT_RX_QUEUES:
+      g_value_set_uint(value, src->devArgs.rx_queues_cnt[MTL_PORT_P]);
+      break;
+    case PROP_RX_FRAMEBUFF_NUM:
+      g_value_set_uint(value, src->framebuffer_num);
+      break;
+    case PROP_RX_CHANNEL:
+      g_value_set_uint(value, src->channel);
+      break;
+    case PROP_RX_SAMPLING:
+      g_value_set_uint(value, src->sampling);
+      break;
+    case PROP_RX_AUDIO_FORMAT:
+      g_value_set_string(value, src->audio_format);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+      break;
+  }
+}
+
+/*
+ * Create MTL session rx handle and initialize the session with the parameters
+ * from caps negotiated by the pipeline.
+ */
+
+static gboolean gst_mtl_st30p_rx_negotiate(GstBaseSrc* basesrc) {
+  GstAudioInfo* info;
+  Gst_Mtl_St30p_Rx* src = GST_MTL_ST30P_RX(basesrc);
+  struct st30p_rx_ops* ops_rx = &src->ops_rx;
+  gint ret;
+  GstCaps* caps;
+
+  info = gst_audio_info_new();
+
+  switch (ops_rx->fmt) {
+    case ST30_FMT_PCM24:
+      gst_audio_info_set_format(info, GST_AUDIO_FORMAT_S24LE, info->rate, info->channels,
+                                NULL);
+      break;
+    case ST30_FMT_PCM16:
+      gst_audio_info_set_format(info, GST_AUDIO_FORMAT_S16LE, info->rate, info->channels,
+                                NULL);
+      break;
+    case ST30_FMT_PCM8:
+      gst_audio_info_set_format(info, GST_AUDIO_FORMAT_S8, info->rate, info->channels,
+                                NULL);
+      break;
+    default:
+      GST_ERROR("Unsupported pixel format %d", ops_rx->fmt);
+      return FALSE;
+  }
+
+  info->rate = ops_rx->sampling;
+  info->channels = ops_rx->channel;
+
+  caps = gst_caps_new_simple("audio/x-raw", "format", G_TYPE_STRING,
+                             gst_audio_format_to_string(info->finfo->format), "channels",
+                             G_TYPE_INT, info->channels, "rate", G_TYPE_INT, info->rate,
+                             NULL);
+
+  if (!caps) caps = gst_pad_get_pad_template_caps(GST_BASE_SRC_PAD(basesrc));
+
+  if (gst_caps_is_empty(caps)) {
+    gst_caps_unref(caps);
+    return FALSE;
+  }
+
+  ret = gst_pad_set_caps(GST_BASE_SRC_PAD(basesrc), caps);
+  gst_caps_unref(caps);
+  if (!ret) {
+    GST_ERROR("Failed to set caps");
+    return FALSE;
+  }
+
+  gst_audio_info_free(info);
```

Comment:
info isn't freed every time when this function returns.

Reply:
Fixed